src/libraries/System.Private.CoreLib/src/System/Threading/NativeRuntimeEventSource.PortableThreadPool.cs

```diff
@@ -1,28 +1,20 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Threading;
 using System.Diagnostics.Tracing;
 using System.Runtime.CompilerServices;
 using Internal.Runtime.CompilerServices;
 
-namespace System.Threading
+namespace System.Diagnostics.Tracing
 {
-    // Currently with EventPipe there isn't a way to move events from the native side to the managed side and get the same
-    // experience. For now, the same provider name and guid are used as the native side and a temporary change has been made to
-    // EventPipe in CoreCLR to get thread pool events in performance profiles when the portable thread pool is enabled, as that
-    // seems to be the easiest way currently and the closest to the experience when the portable thread pool is disabled.
-    // TODO: Long-term options (also see https://github.com/dotnet/runtime/issues/38763):
-    // - Use NativeRuntimeEventSource instead, change its guid to match the provider guid from the native side, and fix the
-    //   underlying issues such that duplicate events are not sent. This should get the same experience as sending events from
-    //   the native side, and would allow easily moving other events from the native side to the managed side in the future if
-    //   necessary.
-    // - Use a different provider name and guid (maybe "System.Threading.ThreadPool"), update PerfView and dotnet-trace to
-    //   enable the provider by default when the Threading or other ThreadPool-related keywords are specified for the runtime
-    //   provider, and update PerfView with a trace event parser for the new provider so that it knows about the events and may
-    //   use them to identify thread pool threads.
-    [EventSource(Name = "Microsoft-Windows-DotNETRuntime", Guid = "e13c0d23-ccbc-4e12-931b-d9cc2eee27e4")]
-    [EventSourceAutoGenerate]
-    internal sealed partial class PortableThreadPoolEventSource : EventSource
+    // This is part of the NativeRuntimeEventsource, which is the managed version of the Microsoft-Windows-DotNETRuntime provider.
+    // It contains the handwritten implementation of the ThreadPool events.
+    // The events here do not call into the typical WriteEvent* APIs unlike most EventSources.
+    // Instead, they call directly into QCalls provided by the runtime (refer to XplatEventLogger.cs) which call
+    // FireEtw* methods auto-generated from ClrEtwAll.man. This ensures that corresponding event sinks are being used
+    // for the native platform. Refer to src/coreclr/vm/nativeruntimesource.cpp.
```

Comment:
I would clarify in the comment why WriteEvent() APIs are not being used here, that it causes double logging because of special case handling of this source in the EventListener code and gives us strongly typed LTTng events rather than the default LTTng handling that other EventSources get.